soh/src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -219,11 +219,12 @@ void Randomizer_GiveSkullReward(EnSi* this, GlobalContext* globalCtx) {
 
     if (getItem.modIndex == MOD_NONE) {
         Item_Give(globalCtx, giveItemId);
+
+        // RANDOTOD: Move this into Item_Give() or some other more central location
+        if (getItem.getItemId == GI_SWORD_BGS) {
+            gSaveContext.bgsFlag = true;
+        }
```

Comment:
This should probably go above the `Item_Give` but I'm not 100% sure if it matters.

Reply:
i can move it, it was below when it was outside of the `MOD_NONE` check so i figured it'd be fine

Reply:
I copied this pattern from this file to shopsanity, so we might need to make this adjustment there too

Reply:
sounds good, i'll try moving the check above item give and testing that everything works as expected, and then i'll do the same for shops